libs/coin-evm/src/api/gasTracker/index.ts

```diff
@@ -0,0 +1,19 @@
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { GasOptions } from "../../types";
+import { getGasOptions as ledgerGetGasOptions } from "./ledger";
+
+export const getGasTracker = (
+  currency: CryptoCurrency
+): {
+  getGasOptions: (currency: CryptoCurrency) => Promise<GasOptions>;
+} => {
+  switch (currency.ethereumLikeInfo?.gasTracker?.type) {
+    case "ledger":
+      return {
+        getGasOptions: ledgerGetGasOptions,
+      };
+
+    default:
+      throw new Error(`No gas tracker found for currency "${currency.id}"`);
+  }
```

Comment:
That might be problematic with the React hook, as that means you'd need to have a conditional hook or a try/catch in it for all the currencies that won't have a gasTracker 🤔